### PR TITLE
Added mpHealth-3.1 FAT

### LINKED ISSUE #17314: Add additional MpHealth-3.1 FAT
- Add tests for application startup states in Open Liberty for /health/start
- Add tests MPConfig property `mp.health.default.startup.empty.response`
- Repeat existing MP Health-2.0 FATs for mpHealth-3.1

### PR DESCRIPTION
**Pull request by pgunapal**
fixes #17314

- Add tests for application startup states in Open Liberty for /health/start
- Add tests MPConfig property mp.health.default.startup.empty.response
- Repeat existing MP Health-2.0 FATs for mpHealth-3.1

### CODE REVIEW

**Comment by pgunapal** [comment]
#build

**Comment by pgunapal** [comment]
#build

File: dev/io.openliberty.microprofile.health.3.1.internal_fat/fat/src/io/openliberty/microprofile/health31/fat/DefaultOverallStartupStatusUpAppStartupTest.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    public void setupClass(LibertyServer server, String testName) throws Exception {
+        log("setupClass", testName + " - Deploying the Delayed App into the apps directory and starting the server.");
+
+        WebArchive app = ShrinkHelper.buildDefaultApp(APP_NAME, "io.openliberty.microprofile.health31.delayed.health.check.app");
+        ShrinkHelper.exportAppToServer(server, app, DeployOptions.DISABLE_VALIDATION);
+
+        if (!server.isStarted())
+            server.startServer(false, false);
+
+        server.waitForStringInLog("CWWKT0016I: Web application available.*DelayedHealthCheckApp*");
```
Comment (fmhwong) [inline dev/io.openliberty.microprofile.health.3.1.internal_fat/fat/src/io/openliberty/microprofile/health31/fat/DefaultOverallStartupStatusUpAppStartupTest.java]: Do we need an assert for CWWKT0016I not found?

File: dev/io.openliberty.microprofile.health.3.1.internal_fat/fat/src/io/openliberty/microprofile/health31/fat/SlowAppStartupHealthCheckTest.java [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+        log("setupClass", testName + " - Deploying the Delayed App into the apps directory and starting the server.");
+
+        WebArchive app = ShrinkHelper.buildDefaultApp(APP_NAME, "io.openliberty.microprofile.health31.delayed.health.check.app");
+        //This test expects to hit the server before the app is started so we disable validation to prevent the test framework waiting for the app to start.
+        ShrinkHelper.exportAppToServer(server, app, DeployOptions.DISABLE_VALIDATION);
+
+        if (!server.isStarted())
+            server.startServer();
+
+        server.waitForStringInLog("CWWKT0016I: Web application available.*DelayedHealthCheckApp*");
```
Comment (fmhwong) [inline dev/io.openliberty.microprofile.health.3.1.internal_fat/fat/src/io/openliberty/microprofile/health31/fat/SlowAppStartupHealthCheckTest.java]: Do we need an assert for CWWKT0016I not found?

File: dev/io.openliberty.microprofile.health.3.1.internal_fat/fat/src/io/openliberty/microprofile/health31/fat/DefaultOverallStartupStatusUpAppStartupTest.java (hunk 1 above)
Comment (pgunapal) [reply dev/io.openliberty.microprofile.health.3.1.internal_fat/fat/src/io/openliberty/microprofile/health31/fat/DefaultOverallStartupStatusUpAppStartupTest.java]: Updated.

File: dev/io.openliberty.microprofile.health.3.1.internal_fat/fat/src/io/openliberty/microprofile/health31/fat/SlowAppStartupHealthCheckTest.java (hunk 2 above)
Comment (pgunapal) [reply dev/io.openliberty.microprofile.health.3.1.internal_fat/fat/src/io/openliberty/microprofile/health31/fat/SlowAppStartupHealthCheckTest.java]: Updated.

**Comment by pgunapal** [comment]
#build

**Review by fmhwong — APPROVED** [review]
LGTM

**Comment by pgunapal** [comment]
#libby

**Pull request merged**